src/Raven.Client/Documents/Session/Operations/StreamOperation.cs

```diff
@@ -112,9 +114,13 @@ public IEnumerator<BlittableJsonReaderObject> SetResult(StreamResult response)
                     UnmanagedJsonParserHelper.ThrowInvalidJson();
 
                 var property = UnmanagedJsonParserHelper.ReadString(_session.Context, response.Stream, parser, state, buffer);
-                if (string.Equals(property, "Results") == false)
-                    UnmanagedJsonParserHelper.ThrowInvalidJson();
 
+                if (string.Equals(property, nameof(StreamQueryStatistics.ResultEtag)))
```

Comment:
Do not check if property is ResultEtag, do it inside the EnsureValidStreamQueryStats